Automattic/wp-calypso
# Segmentation Survey: Fix saving answers error when leaving the segmentation survey step

Closes https://github.com/Automattic/wp-calypso/issues/89851
Closes https://github.com/Automattic/wp-calypso/issues/90040
Closes https://github.com/Automattic/wp-calypso/issues/90041
Closes https://github.com/Automattic/wp-calypso/issues/90042

## Proposed Changes

* Extracted the `SegmentationSurvey` component and navigation functions
  * Removed the dependency from the `SurveyContext` to make it easier to use
  * Moved functions from the provider to a hook, also removing the NavigationControls dependency
  * Added a surveyKey prop to the main component
* Fixed the saving issue due to the early routing when moving to the next step
* Disabled clickable items when the answers are being saved
* Sent 'skip' keyword as an answer key for skipped or empty questions
* Removed feature flag
* Logged save mutation errors

## Testing Instructions

* Apply this PR to your local
* Go to http://calypso.localhost:3000/setup/entrepreneur/start
* Verify if the survey was rendered as expected
* Open the Network tab on Dev Tools to verify if the questions are being saved
* Answer the first question and click Continue
* All the clickable items should be disabled while the request is in progress
* Submit the second question and verify if the answer was saved
* Reopen http://calypso.localhost:3000/setup/entrepreneur/start
  * The survey should be rendered without preloaded answers (localStorage cleaned)
* Perform regression tests on tracks events
  * Check if failed requests are correctly tracked
* Perform regression tests on saving skipped questions

## Pre-merge Checklist
